src/modules.js

```diff
@@ -213,39 +235,60 @@ if (__analytics.amplitude?.isEnabled) {
         analyticsApi.defaults.headers.Authorization = `Bearer ${token}`;
         await analyticsApi.post(publishURL, payload);
       };
-      initializeApp(firebaseConfig);
-      const auth = getAuth();
-      signInAnonymously(auth);
-      let eventQueue = [];
+
       const getUid = () => (__analytics.infermedicaAnalytics?.sendUID
         ? firebaseUser.uid
         : null);
-      onAuthStateChanged(auth, async (user) => {
-        if (!user) return;
-        firebaseUser = user;
-        eventQueue.forEach((event) => {
-          const { user } = event;
-          publish({
-            ...event,
-            user: {
-              ...user,
-              id: getUid(),
-            },
-          });
-        });
-        eventQueue = [];
-      });
 
       return {
         name: names.INFERMEDICA_ANALYTICS,
-        trackEvent(eventName, properties) {
+        /**
+         * @param {import('./main').IInitializeParams} options
+         */
+        initialize: async (options) => {
+          auth = options.firebaseAuth;
+
+          onAuthStateChanged(auth, async (authUser) => {
```

Comment:
options.firebaseAuth is defined as optional in IInitializeParams. If auth var will be undefined this will work?

Reply:
Ok, I've changed this ;)